components/common/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,62 @@
+import { faDownload } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect, useState } from "react";
+
+interface BeforeInstallPromptEvent extends Event {
+  readonly platforms: string[];
+  readonly userChoice: Promise<{
+    outcome: "accepted" | "dismissed";
+    platform: string;
+  }>;
+  prompt(): Promise<void>;
+}
+
+declare global {
+  interface WindowEventMap {
+    beforeinstallprompt: BeforeInstallPromptEvent;
+  }
+}
+
+export function Navbar() {
+  const [showButton, setShowButton] = useState(false);
+  const [deferredPrompt, setDeferredPrompt] =
+    useState<BeforeInstallPromptEvent | null>(null);
+
+  if (typeof window !== "undefined") {
+    console.log("test");
+    window.addEventListener("beforeinstallprompt", (e) => {
```

Comment:
> The beforeinstallprompt event fires on devices when a user is about to be prompted to "install" a web application. It may be saved for later and used to prompt the user at a more suitable time.

Looking at the documentation, this event fires when user is about to be prompted right? What triggers the prompt on the first place?

Reply:
It seems we don't fulfill below condition?

- The PWA must not already be installed
- Meets a user engagement heuristic (previously, the user had to interact with the domain for at least 30 seconds, this is not a requirement anymore).
- Your web app must include a web app manifest.
- Your web app must be served over a secure HTTPS connection.
- Has registered a service worker with a fetch event handler.

Reply:
yup, just checked the condition for installment. I tried using different browser (with no PWA installed on the device) and it still doesn't show if you didn't engage with the domain, but it works well after i interact with it for couple seconds